x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/components/steps/step_define_package_policy.tsx

```diff
@@ -86,18 +86,6 @@ export const StepDefinePackagePolicy: React.FunctionComponent<{
       });
     }
 
-    // Update package policy's package and agent policy info
-    useEffect(() => {
-      //  TODO move this to parent hook
-      // If agent policy has changed, update package policy's agent policy ID and namespace
-      if (agentPolicy && packagePolicy.policy_id !== agentPolicy.id) {
-        updatePackagePolicy({
-          policy_id: agentPolicy.id,
-          namespace: agentPolicy.namespace,
-        });
-      }
-    }, [packagePolicy, agentPolicy, packageInfo, updatePackagePolicy]);
-
```

Comment:
this effect was lifted to a single parent hook - `onSubmit`, which is 1 of 3 usages. the other two are not relevant:
- edit screen doesn't allow changing agent policy
- multi page layout doesn't use `package-policy-replace-define-step`


alternatively, we can instead keep it as is and duplicate it in our custom extension component.